Apps/GatewayApi/test/unit/Services.Test/UserProfileServiceTests.cs

```diff
@@ -17,726 +17,1235 @@ namespace HealthGateway.GatewayApiTests.Services.Test
+        /// ValidateMinimumAgeAsync.
         /// </summary>
+        /// <param name="age">The age to validate.</param>
+        /// <param name="minAge">The minimum age to validate against.</param>
+        /// <param name="patientErrorExists">The value indicating whether patient error exists or not.</param>
         /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
-        [Fact]
-        public async Task ShouldValidateAgeAsync()
+        [InlineData(0, 0, false)]
+        [InlineData(19, 19, false)]
+        [InlineData(20, 19, false)]
+        [Theory]
+        public async Task ShouldValidateMinimumAgeAsync(int age, int minAge, bool patientErrorExists)
         {
             // Arrange
-            PatientModel patientModel = new()
-            {
-                Birthdate = DateTime.Now.AddYears(-15),
-            };
-
-            Dictionary<string, string?> localConfig = new()
-            {
-                { "WebClient:MinPatientAge", "19" },
-            };
-
-            UserProfileServiceMock mockService = new UserProfileServiceMock(GetIConfigurationRoot(localConfig))
-                .SetupPatientServiceMockCustomPatient(this.hdid, patientModel);
-            IUserProfileService service = mockService.UserProfileServiceMockInstance();
-            RequestResult<bool> expected = new() { ResultStatus = ResultType.Success, ResourcePayload = false };
+            ValidateMinimumAgeMock mock = SetupValidateMinimumAgeMock(age, minAge, patientErrorExists, true);
 
             // Act
-            RequestResult<bool> actualResult = await service.ValidateMinimumAgeAsync(this.hdid);
+            RequestResult<bool> actual = await mock.Service.ValidateMinimumAgeAsync(mock.Age);
 
             // Assert
-            Assert.Equal(ResultType.Success, actualResult.ResultStatus);
-            Assert.Equal(expected.ResourcePayload, actualResult.ResourcePayload);
+            mock.Expected.ShouldDeepEqual(actual);
         }
 
         /// <summary>
-        ///  CreateUserPreference call.
+        /// ValidateMinimumAgeAsync fails validation due to invalid age.
         /// </summary>
-        /// <param name="dBStatusCode">dBStatusCode.</param>
+        /// <param name="age">The age to validate.</param>
+        /// <param name="minAge">The minimum age to validate against.</param>
+        /// <param name="patientErrorExists">The value indicating whether patient error exists or not.</param>
         /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [InlineData(0, 19, false)]
+        [InlineData(18, 19, false)]
+        [InlineData(19, 19, true)]
         [Theory]
-        [InlineData(DbStatusCode.Created)]
-        [InlineData(DbStatusCode.Error)]
-        public async Task ShouldCreateUserPreference(DbStatusCode dBStatusCode)
+        public async Task ShouldNotValidateInvalidMinimumAgeAsync(int age, int minAge, bool patientErrorExists)
```

Comment:
Similar naming issue here - validation implies both accepting valid inputs and rejecting invalid ones, so the test shouldn't be to "not validate".

Again, I'd suggest combining with the previous test - `ShouldValidateMinimumAgeAsync` - or renaming both:

- `ShouldValidateValidMinimumAge` and `ShouldValidateInvalidMinimumAge`, or
- `ShouldValidateMinimumAgeSuccess` and `ShouldValidateMinimumAgeFailure`, or
- `ShouldReturnSuccessForValidateMinimumAgeAsync` and `ShouldReturnFailureForValidateMinimumAgeAsync`, or
- `ValidateMinimumAgeAsyncShouldReturnSuccess` and `ValidateMinimumAgeAsyncShouldReturnFailure`

Reply:
Changed names to ShouldValidateValidMinimumAge and ShouldValidateInvalidMinimumAge